src/data/s3-data-source.ts

```diff
@@ -92,12 +114,15 @@ export class S3DataSource implements ContiguousDataSource {
       if (response.Body === undefined) {
         throw new Error('Body missing from S3 response');
       }
+      const payloadContentTypeS3MetaDataTag = 'x-amz-meta-payload-content-type';
+      const sourceContentType =
+        head.headers?.[payloadContentTypeS3MetaDataTag] ?? response.ContentType;
 
       return {
         stream: response.Body as Readable,
         size: response.ContentLength,
```

Comment:
@fedellen @djwhitt don't we use this size when indexing a bundle? If we pass the size of the range we're getting it can mess with the indexed bundle total size, right?

Reply:
Yeah, @karlprieb is right. The size here should be the size of the data as opposed to the entire content length.

Reply:
the response of the getRequest with a range request is the size of the data payload here. Just added some debug logs there

<img width="1207" alt="image" src="https://github.com/ar-io/ar-io-node/assets/60207215/3fa14929-5713-435c-a494-309c386cef72">

Reply:
Cool. I may also have misread @karlprieb's comment too. It might have been intended to say the reverse (the size is too small). That's actually not a problem. The size in this context is always data size as opposed to raw data item size.